logstash-core/src/main/java/org/logstash/FileLockFactory.java

```diff
@@ -39,7 +41,8 @@
 
     private FileLockFactory() {}
 
-    private static final Set<String> LOCK_HELD = Collections.synchronizedSet(new HashSet<String>());
+    private static final Set<String> LOCK_HELD = Collections.synchronizedSet(new HashSet<>());
```

Comment:
that's not it  -
1- you obtain a lock by specifying a lock dir and a lock name, this returns a FileLock object. 
2- you release a lock by passing back the previously returned FileLock object hence the need for both LOCK_HELD and LOCK_MAP.  
- you want to only be able to release a lock which was first obtained.